app/models/user.rb

```diff
@@ -1660,6 +1660,14 @@ def disable_redesigned_user_menu
     Discourse.redis.del("#{REDESIGN_USER_MENU_REDIS_KEY_PREFIX}#{self.id}")
   end
 
+  def sidebar_categories_ids
+    return category_sidebar_section_links.pluck(:linkable_id) if category_sidebar_section_links.present?
```

Comment:
Doing `category_sidebar_section_links.present?` will actually load the associations which is wasteful considering that we only need the ids. I think we can just do a pluck regardless.